notes/notes.proto

```diff
@@ -88,22 +88,31 @@ message ListNotesReply {
 }
 
 message AddBlockRequest {
+    //the note id where we add a block
+    string note_id = 1;
+
     // The block to append to the note.
-    Block block = 1;
+    Block block = 2;
 
     // Given that a note is an array of blocks, provide the index at
     // which the block shall be inserted.
-    uint32 index = 2;
+    uint32 index = 3;
 }
 
 message UpdateBlockRequest {
-    string id = 1;
+    // The note id where we add a block
+    string note_id = 1;
+    
+    Block block = 3;
 
-    Block block = 2;
+    // In order to move the position of an existing block
+    uint32 index = 4;
 
-    google.protobuf.FieldMask update_mask = 3;
+    google.protobuf.FieldMask update_mask = 5;
 }
 
 message DeleteBlockRequest {
-    string id = 1;
+    string note_id = 1;
```

Comment:
Mhm, do we really need this? a `Block.id` is supposed to be unique across all notes right (aka `Block.id` is sufficient to uniquely identify a block)? Only reason why we might need it is to simplify the authorisation process.